orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/CreateServerGroupStage.groovy

```diff
@@ -43,11 +48,28 @@ class CreateServerGroupStage extends AbstractDeployStrategyStage {
   protected List<TaskNode.TaskDefinition> basicTasks(Stage stage) {
     def taggingEnabled = featuresService.isStageAvailable("upsertEntityTags")
 
-    def tasks = [
-      TaskNode.task("createServerGroup", CreateServerGroupTask),
-      TaskNode.task("monitorDeploy", MonitorKatoTask),
-      TaskNode.task("forceCacheRefresh", ServerGroupCacheForceRefreshTask),
-    ]
+    def tasks = []
+    String kind = stage.context?.get("kind")
+
+    if (stage.context?.get("kind").toString() == KUBERNETES_STATEFULSET_CONTROLLER_TYPE ||
```

Comment:
I don't think that this is the way to handle this. These stages are meant to be generic, and putting top level references to k8s resources makes this code fairly messy.